include/rocksdb/advanced_options.h

```diff
@@ -320,6 +320,23 @@ struct AdvancedColumnFamilyOptions {
   // Dynamically changeable through SetOptions() API
   size_t inplace_update_num_locks = 10000;
 
+  // [experimental]
+  // Used to activate or deactive the Mempurge feature (memtable garbage
+  // collection). (deactivated by default). At every flush, the total useful
+  // payload (total entries minus garbage entries) is estimated as a ratio
+  // [useful payload bytes]/[size of a memtable (in bytes)]. This ratio is then
+  // compared to this `threshold` value:
+  //     - if ratio<threshold: the flush is replaced by a mempurge operation
+  //     - else: a regular flush operation takes place.
+  // Threshold values:
+  //   0.0: mempurge deactivated (default).
+  //   1.0: recommended threshold value.
+  //   >1.0 : aggressive mempurge.
+  //   0 < threshold < 1.0: mempurge triggered only for very low useful payload
```

Comment:
Is there any code or tests that does anything other than 0 or > 0 now?  I might have missed it

Reply:
Technically there is nothing wrong with using a threshold < 0, but it would be strictly equivalent to using 0, because this threshold value is then compared to an estimated garbage ratio obtained from sampling (see [here](https://github.com/facebook/rocksdb/blob/dde774db646d74054a312878244206431ee939e4/db/flush_job.cc#L634)). 
Overall you are bringing up a valid point, I will include a couple tests where the value is < 0 just for the sake of safety.

Reply:
I see what you mean. For these unit tests specifically I cannot use values between 0.0 and 1.0 (e.g.: I cannot use 0.5) because the behavior is non-deterministic (it's compared to a sampled value). I ran bench tests a while ago that showed consistent results with the threshold values. Otherwise crash tests use any values between 0 and 10.0 (see [here](https://github.com/facebook/rocksdb/blob/dde774db646d74054a312878244206431ee939e4/tools/db_crashtest.py#L271)
However following your suggestion I will replace some "0.0" by values <0.0 and some "1.0" by values >1.0 in the unit tests. Thanks for pointing that out.